### ISSUE TITLE
Tor completely blocked by YouTube

### ISSUE THREAD

**Issue by user234683** [user234683/youtube-local#134]
All Tor requests are now blocked. However, visiting YouTube via the Tor Browser is *not* blocked. After having youtube-local use the same exit nodes and same headers, the main difference I see is the usage of http2 in Tor Browser. I do not have time for further investigation than this until mid June.

If anyone can figure out if HTTP2 is the thing that will bypass the restriction, that would be appreciated. [HTTP2 requests in Python](https://stackoverflow.com/questions/34570991/http-2-request-in-python-2-7).

Test url to visit using a Python script, routing through Tor: `https://www.youtube.com/results?search_query=test`

**Comment by user234683** [comment]
> > All Tor requests are now blocked.
> 
> I cannot confirm this. The "On, except video" setting has been working well for me, on 2 different machines (same IP though).
> 
> Tor version:
> 
> ```
> Tor version 0.4.7.7.
> Tor is running on Linux with Libevent 2.1.12-stable, OpenSSL 1.1.1o, Zlib 1.2.12, Liblzma 5.2.5, Libzstd 1.5.2 and Glibc 2.35 as libc.
> Tor compiled with GCC version 12.1.0
> ```
> 
> and
> 
> ```
> Tor version 0.4.5.10.
> ```
> 
> Again, both are working well enough, except when I was testing a new install and apparently overdid it. I'm running tor without entry/exit node restrictions, and the necessary settings to make it work with this repo.
> 
> PS: is the tor software changing circuits automatically at regular intervals? Or does youtube-local implement this?

Ah, working on my end now as well. Looks like I left an ExitNodes restriction in there while debugging this. Yes, Tor changes circuits on its own periodically. youtube-local only tries to change circuits when the exit node is blocked

**Comment by user234683** [comment]
As of 1/20/2024, Tor routing appears to still work, so closing this issue. Feel free to open another issue if there are recent problems
